src/components/NewAnalysisModal.js

```diff
@@ -64,44 +133,135 @@ export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
       })
     })
 
-    const NEW_JUPYTER_MODE = tools.Jupyter.label
-    const NEW_RSTUDIO_MODE = tools.RStudio.label
-    const NEW_GALAXY_MODE = tools.galaxy.label
-    const getView = () => Utils.switchCase(viewMode,
-      [NEW_JUPYTER_MODE, () => renderNewRuntimeModal(NEW_JUPYTER_MODE)],
-      [NEW_RSTUDIO_MODE, () => renderNewRuntimeModal(NEW_RSTUDIO_MODE)],
-      [NEW_GALAXY_MODE, renderNewGalaxyModal],
-      [Utils.DEFAULT, renderCreateAnalysisBody]
-    )
-
-    const toolCardStyles = { backgroundColor: 'white', borderRadius: 5, padding: '1rem', display: 'inline-block', verticalAlign: 'middle', marginBottom: '1rem', textAlign: 'center', width: '100%', height: 60 }
-    const imageStyles = { verticalAlign: 'middle', height: '100%', width: '40%' }
+    const styles = {
+      toolCard: { backgroundColor: 'white', borderRadius: 5, padding: '1rem', display: 'inline-block', verticalAlign: 'middle', marginBottom: '1rem', textAlign: 'center', width: '100%', height: 60 },
+      image: { verticalAlign: 'middle', height: '100%', width: '40%' }
+    }
 
     const renderToolButtons = () => div({ style: { display: 'flex', alignItems: 'center', flexDirection: 'column', justifyContent: 'space-between' } }, [
-      div({ style: toolCardStyles, onClick: () => setViewMode(NEW_JUPYTER_MODE) }, [img({ src: jupyterLogoLong, style: _.merge(imageStyles, { width: '30%' }) })]),
-      div({ style: toolCardStyles, onClick: () => setViewMode(NEW_RSTUDIO_MODE) }, [img({ src: rstudioLogo, style: imageStyles })]),
-      div({ style: toolCardStyles, onClick: () => setViewMode(NEW_GALAXY_MODE) }, [img({ src: galaxyLogo, style: _.merge(imageStyles, { width: '30%' }) })])
+      div({ style: styles.toolCard, onClick: () => { setCurrentTool(tools.Jupyter.label); enterNextViewMode(tools.Jupyter.label) } }, [img({ src: jupyterLogoLong, style: _.merge(styles.image, { width: '30%' }) })]),
+      div({ style: styles.toolCard, onClick: () => { setCurrentTool(tools.RStudio.label); enterNextViewMode(tools.RStudio.label) } }, [img({ src: rstudioLogo, style: styles.image })]),
+      div({ style: { opacity: currentApp ? '0.5' : '1', ...styles.toolCard }, onClick: () => { setCurrentTool(tools.galaxy.label); enterNextViewMode(tools.galaxy.label) }, disabled: !currentApp, title: currentApp ? 'You already have a galaxy environment' : '' }, [img({ src: galaxyLogo, style: _.merge(styles.image, { width: '30%' }) })])
```

Comment:
Nit: wrap. We're currently debating a line limit of 100 or 120 characters. For now, please just treat 120 as a soft limit.